Week1/homework/app.js

```diff
@@ -1,11 +1,169 @@
+      language: 'English',
+      author: 'By J. R. R. Tolkien',
+    },
+    'the-hobbit': {
+      title: 'The Hobbit',
+      cover: 'https://upload.wikimedia.org/wikipedia/en/4/4a/TheHobbit_FirstEdition.jpg',
+      language: 'English',
+      author: 'By J. R. R. Tolkien',
+    },
+    'a-game-of-thrones': {
+      title: 'A Game of Thrones',
+      cover: 'https://upload.wikimedia.org/wikipedia/en/9/93/AGameOfThrones.jpg',
+      language: 'English',
+      author: 'By George R. R. Martin',
+    },
+    'harry-potter-and-the-goblet-of-fire': {
+      title: 'Harry Potter and the Goblet of Fire',
+      cover:
+        'https://upload.wikimedia.org/wikipedia/en/c/c7/Harry_Potter_and_the_Goblet_of_Fire.jpg',
+      language: 'English',
+      author: 'By J. K. Rowling',
+    },
+    'the-book-thief': {
+      title: 'The Book Thief',
+      cover:
+        'https://upload.wikimedia.org/wikipedia/en/8/8f/The_Book_Thief_by_Markus_Zusak_book_cover.jpg',
+      language: 'English',
+      author: 'By Markus Zusak',
+    },
+    'angels-and-demons': {
+      title: 'Angels and Demons',
+      cover: 'https://upload.wikimedia.org/wikipedia/en/5/5f/AngelsAndDemons.jpg',
+      language: 'English',
+      author: 'By Dan Brown',
+    },
+    'harry-potter-and-the-order-of-the-phoenix': {
+      title: 'Harry Potter and the Order of the Phoenix',
+      cover:
+        'https://upload.wikimedia.org/wikipedia/en/7/70/Harry_Potter_and_the_Order_of_the_Phoenix.jpg',
+      language: 'English',
+      author: 'By J. K. Rowling',
+    },
+  };
+
+  // I created another function, which takes books object as a parameter  Object.keys() method returns an array of a given object's own property names
+  // in the same order as we get with a normal loop. After getting array of object names, I used forEach method which is
+  // an array method to iterate through the names in the array.
+  // Inside forEach method, I used The Document method getElementById()
+  // that returns an Element object representing the element whose id property matches the specified string.
+  // Then created div element by using document.createElement() method, and used setAttribute method to set class attribute and its value on div.
+  // Here I wanted to be DRY, I created a re-useable function to create and append and set attribute.
+  // Here I used element[property] bracket notation, since I'm using the string passed to the property parameter.
+  // if you use dot notation JS thinks that I'm looking for a property that its name is property.
+  // Used if statement to set alt attribute only when the elementName is img.
+  // For how to use bracket notation:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors
+
+  const listOfBooks = obj =>
+    Object.keys(obj).forEach(item => {
+      const booksContainer = document.getElementById('books-container');
+
+      const bookDiv = document.createElement('div');
```

Comment:
Although it works, semantically speaking, having a `div` tag acting as a list container instead of `ul` isn't the best idea.

Reply:
Also, `bookDiv` isn't the best of variable names. How about `booksContainer` because that is what the element acts like, isn't it?

Reply:
I created a new list item and appended to the ul element!